contracts/truefi2/TrueCreditAgency.sol

```diff
@@ -129,11 +131,14 @@ contract TrueCreditAgency is UpgradeableClaimable {
     }
 
     function poke(ITrueFiPool2 pool) public {
-        for (uint16 i = 1; i <= MAX_CREDIT_SCORE; i++) {
-            CreditScoreBucket storage bucket = buckets[pool][i];
-            if (bucket.borrowersCount == 0) {
+        uint256 bitMap = usedBuckets;
+        for (uint16 i = 0; i <= MAX_CREDIT_SCORE; (i++, bitMap >>= 1)) {
+            if (bitMap & 1 == 0) {
                 continue;
             }
+
+            CreditScoreBucket storage bucket = buckets[pool][i];
```

Comment:
Something to A/B test in a separate PR: can we look up `buckets[pool]` only once? Otherwise we'd call it once for every used bucket in this loop.

Reply:
Surprisingly it reduced the gas consumption just by a tiny bit (383741 vs 385374) for a large amount of buckets. Solidity must be doing some internal optimization. But still a good catch, will take it out of loop

Reply:
Surprisingly it reduced the gas consumption just by a tiny bit (383741 vs 385374) for a large amount of buckets (20). Solidity must be doing some internal optimization. But still a good catch, will take it out of loop